src/parser/jobs/rpr/modules/Gauge.tsx

```diff
@@ -0,0 +1,197 @@
+import {Trans} from '@lingui/react'
+import Color from 'color'
+import {DataLink} from 'components/ui/DbLink'
+import {ActionKey} from 'data/ACTIONS'
+import {Event, Events} from 'event'
+import {filter} from 'parser/core/filter'
+import {dependency} from 'parser/core/Injectable'
+import {CounterGauge, Gauge as CoreGauge} from 'parser/core/modules/Gauge'
+import Suggestions, {SEVERITY, TieredSuggestion} from 'parser/core/modules/Suggestions'
+import React from 'react'
+
+const SOUL_MAX_VALUE = 100
+const SHROUD_MAX_VALUE = 100
+
+const SOUL_GAUGE_COLOR = Color('#DC0707FF')
+const SHROUD_GAUGE_COLOR = Color('#2C9FCB')
+
+const SOUL_GENERATORS: ActionKey[] = [
+	'SLICE',
+	'WAXING_SLICE',
+	'INFERNAL_SLICE',
+	'SPINNING_SCYTHE',
+	'NIGHTMARE_SCYTHE',
+	'SOUL_SLICE',
+	'SOUL_SCYTHE',
+]
```

Comment:
I would spin Soul Slice and Soul Scythe out into their own ActionKey[], since it's usually better to handle defaults and edge cases, than to handle everything with special logic.